bpf/process/types/basic.h

```diff
@@ -631,9 +632,101 @@ copy_char_buf(void *ctx, long off, unsigned long arg, int argm,
 	return __copy_char_buf(ctx, off, arg, bytes, has_max_data(argm), e, data_heap);
 }
 
+static inline __attribute__((always_inline)) long
+filter_char_buf_equal(struct selector_arg_filter *filter, char *arg_str, uint orig_len)
+{
+	__u32 *map_ids = (__u32 *)&filter->value;
+	void *string_map;
+	__u32 map_idx;
+	__u8 len;
+	__u8 padded_len;
+	int index;
+	char *heap, *zero_heap;
+	int zero = 0;
+
+	if (orig_len >= STRING_MAPS_SIZE_5 || !orig_len)
+		return 0;
+
+	len = (__u8)orig_len;
+	// Calculate padded string length
+	padded_len = len;
+	if (len % STRING_MAPS_KEY_INC_SIZE != 0)
+		padded_len = ((len / STRING_MAPS_KEY_INC_SIZE) + 1) * STRING_MAPS_KEY_INC_SIZE;
```

Comment:
Instead of the branch and the divide, you could also do:
```
padded_len = len  + ((STRING_MAPS_KEY_INC_SIZE - (size % STRING_MAPS_KEY_INC_SIZE)) % STRING_MAPS_KEY_INC_SIZE)
```

Test:

```shell-interactive
$ python3
Python 3.11.5 (main, Aug 29 2023, 15:31:31) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def align1(size, alignment):
...      return size + ((alignment - (size % alignment)) % alignment)
...
>>> def align2(size, alignment):
...      if size % alignment == 0:
...          return size
...      return ( (size  // alignment) + 1)  * alignment
...
>>> for i in range(10000):
...     assert align1(i, 24) == align2(i, 24)
...
>>> align1(24, 24)
24
>>> align1(25, 24)
48

```

Reply:
I think golang will optimise accordingly, and I think the branch and divide is more readable TBH.